app.js

```diff
@@ -72,5 +75,29 @@ if (app.get('env') === 'development') {
 // production error handler
 app.use(errorHandlers.productionErrors);
 
+cron.schedule('0 2 * * * *', async () => {
```

Comment:
This is not the correct way to implement a cron job. Suppose someone wants to update or delete a cron job, then as per this logic, he/she can't do that.

Also, Timing should be added in CONSTS file and then should be used. So that in the future if we want to another run cron job at the same time, then we can use that from CONSTS file. Also, it will be easy to change timings if that is in CONSTS  file.

You can refer to this for cron job management -  https://github.com/dhrumitpatel48/share-memories-MERN/blob/main/server/utils/cron-job-manager.js

Reply:
> This is not the correct way to implement a cron job. Suppose someone wants to update or delete a cron job, then as per this logic, he/she can't do that.

I didn't get your point we can add/update/delete jobs by updating the code. According to me, we need a job manager when we allow the users to run the jobs they want at `run-time` 

> Timing should be added in CONSTS file and then should be used. So that in the future if we want to another run cron job at the same time, then we can use that from CONSTS file. Also, it will be easy to change timings if that is in CONSTS file.

Yeah sure , I will put that in the CONSTS file

> You can refer to this for cron job management - https://github.com/dhrumitpatel48/share-memories-MERN/blob/main/server/utils/cron-job-manager.js

This implementation is good but I think it would be better if we are creating jobs at `run time`. In our case it is fixed that what jobs we want to run so they can be placed in single file like `cron.js` 

@salahlalami @dhrumitpatel48  What do you think?

Reply:
> I didn't get your point we can add/update/delete jobs by updating the code. According to me, we need a job manager when we allow the users to run the jobs they want at `run-time`

My point is there should be general implementation, not specific. Also, implementation should be in such a way that it covers all the cases. If we are creating a job, then there should be the option to update and delete as well.

In future, we don't know what type of requirements will come with respect to  cron jobs

Reply:
> 

I understand your concerns but that is needed if we have dynamic cron jobs. It was nice discussion with you, learned a lot from you about cron jobs with this discussion

Reply:
> This implementation is good but I think it would be better if we are creating jobs at `run time`. In our case, it is fixed that what jobs we want to run so they can be placed in a single file like `cron.js`

As of now, it is fixed but what about the future??
Also if you think that in the future there is a requirement to add cron jobs at multiple places, then will you always write static code???

I think there should be general implementation, and then using that should write specific jobs as per requirements.
In the above case, of course, jobs will be created run time.

In short, I want to convey that write a general JOB MANAGER, then create an instance of it, and then use it wherever you want.

Reply:
> As of now, it is fixed but what about the future?? Also if you think that in the future there is a requirement to add cron jobs at multiple places, then will you always write static code???
> 
> I think there should be general implementation, and then using that should write specific jobs as per requirements. In the above case, of course, jobs will be created run time.
> 
> In short, I want to convey that write a general JOB MANAGER, then create an instance of it, and then use it wherever you want.

What is your view on this @salahlalami ??

Reply:
> > I didn't get your point we can add/update/delete jobs by updating the code. According to me, we need a job manager when we allow the users to run the jobs they want at `run-time`
> 
> My point is there should be general implementation, not specific. Also, implementation should be in such a way that it covers all the cases. If we are creating a job, then there should be the option to update and delete as well.
> 
> In future, we don't know what type of requirements will come with respect to cron jobs

What is your take on this @salahlalami ??

Reply:
> 

@dhrumitpatel48 I am just curious to know that you will in a situation pass the cron job function to the Job manager, So at the end, you will be writing a function for it. So it will be static also. The point is Job manager makes the execution dynamic right?
Feel free to correct me if I am wrong

Reply:
> > 
> 
> @dhrumitpatel48 I am just curious to know that you will in a situation pass the cron job function to the Job manager, So at the end, you will be writing a function for it. So it will be static also. The point is Job manager makes the execution dynamic right? Feel free to correct me if I am wrong

Job Manager should be general. 
Basically, use OOPS and make a job manager.
Then wherever you want to create/update/delete the job, make the instance of the JOB MANAGER class, then use it. Now whenever the server starts, app.js will execute. Cron job will be created.